src/store/aggregator/sagas/withdrawTokens.test.ts

```diff
@@ -0,0 +1,342 @@
+          ...mockSelectors,
+          [matchers.call.fn(mockToken.allowance), parseUnits('11')],
+          [
+            matchers.call.fn(
+              mockMasset['redeemToBridge(address,uint256,address)']
+            ),
+            mockTx,
+          ],
+          [matchers.call.fn(mockTx.wait), mockReceipt],
+        ])
+        .call(
+          mockToken.allowance,
+          mockAccount,
+          mockMasset.address.toLowerCase()
+        )
+        .call(
+          mockMasset['redeemToBridge(address,uint256,address)'],
+          mockBassetAddress,
+          mockAmount,
+          mockReceiver.toLowerCase()
+        )
+        .hasFinalState(expectedState)
+        .run();
+    });
+
+    it('when unsufficient allowance', async () => {
+      const expectedState: AggregatorState = {
+        ...aggregatorInitialState,
+        submitCall: {
+          status: 'success',
+          currentOperation: 'withdraw',
+          steps: [
+            {
+              ...initialSteps?.[1],
+              tx: mockTx,
+              txReceipt: mockReceipt,
+            },
+            {
+              ...initialSteps?.[3],
+              tx: mockTx,
+              txReceipt: mockReceipt,
+            },
+          ],
+        },
+      };
+
+      await expectSaga(
+        withdrawTokens,
+        aggregatorActions.submit(withdrawMockValues)
+      )
+        .withReducer(aggregatorReducer)
+        .withState(aggregatorInitialState)
+        .provide([
+          ...mockSelectors,
+          [matchers.call.fn(mockToken.allowance), parseUnits('4')],
+          [matchers.call.fn(mockToken.approve), mockTx],
+          [
+            matchers.call.fn(
+              mockMasset['redeemToBridge(address,uint256,address)']
+            ),
+            mockTx,
+          ],
+          [matchers.call.fn(mockTx.wait), mockReceipt],
+        ])
+        .call(
+          mockToken.allowance,
+          mockAccount,
+          mockMasset.address.toLowerCase()
+        )
+        .call(mockToken.approve, mockMasset.address.toLowerCase(), mockAmount)
+        .call(
+          mockMasset['redeemToBridge(address,uint256,address)'],
+          mockBassetAddress,
+          mockAmount,
+          mockReceiver.toLowerCase()
+        )
+        .hasFinalState(expectedState)
+        .run();
+    });
+
+    it('when no masset contract', async () => {
+      const expectedState: AggregatorState = {
+        ...aggregatorInitialState,
+        submitCall: {
+          status: 'failure',
+          steps: [
+            {
+              ...initialSteps?.[0],
+              error: 'Could not find contracts',
+            },
+            ...initialSteps.slice(1, 4),
+          ],
+        },
+      };
+
+      await expectSaga(
+        withdrawTokens,
+        aggregatorActions.submit(withdrawMockValues)
+      )
+        .withReducer(aggregatorReducer)
+        .withState(aggregatorInitialState)
+        .provide([
+          ...mockSelectors.slice(0, -1),
```

Comment:
I'm wondering what would happen if it was done this way. Maybe it would override the previous mock, and there would be no need to slice the array?
 
```suggestion
          ...mockSelectors,
```

Reply:
I wanted to remove the last element